dd-sdk-android/src/main/kotlin/com/datadog/android/rum/GlobalRum.kt

```diff
@@ -153,9 +159,31 @@ object GlobalRum {
 
     internal fun updateRumContext(newContext: RumContext) {
         activeContext.set(newContext)
+        updateContextInPlugins(newContext, RumFeature.plugins)
+        updateContextInPlugins(newContext, CrashReportsFeature.plugins)
+        updateContextInPlugins(newContext, LogsFeature.plugins)
+        updateContextInPlugins(newContext, TracesFeature.plugins)
+    }
+
+    private fun updateContextInPlugins(
+        newContext: RumContext,
+        plugins: List<DatadogPlugin>
+    ) {
+        plugins.forEach {
+            it.onContextChanged(
+                DatadogContext(
+                    DatadogRumContext(
+                        newContext.applicationId,
+                        newContext.sessionId,
+                        newContext.viewId
+                    )
+                )
```

Comment:
You're instantiating the exact same object on each iteration here. You could instead create it once (in `updateRumContext` for instance) and pass it to all the plugins.